lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -17,25 +17,21 @@ def initialize(options={})
       # Again, in case we use DEFAULT_OPTIONS in the future, uncomment this.
       # @options = DEFAULT_OPTIONS.merge(options)
       @options = options
-      @client = build_client(@options)
+      @pool = build_pool(@options)
       # mutex to prevent requests and sniffing to access the
       # connection pool at the same time
-      @request_mutex = Mutex.new
-      start_sniffing!
     end
 
     def template_install(name, template, force=false)
-      @request_mutex.synchronize do
-        if template_exists?(name) && !force
-          @logger.debug("Found existing Elasticsearch template. Skipping template management", :name => name)
-          return
-        end
-        template_put(name, template)
+      if template_exists?(name) && !force
+        @logger.debug("Found existing Elasticsearch template. Skipping template management", :name => name)
+        return
       end
+      template_put(name, template)
     end
 
     def bulk(actions)
-      @request_mutex.synchronize { non_threadsafe_bulk(actions) }
+      non_threadsafe_bulk(actions)
```

Comment:
I know this is old code but I think we could just move the code from non_threadsafe_bulk here and document, if necessary, its limitations